osu.Game.Rulesets.Mania/Beatmaps/Patterns/Legacy/HitObjectPatternGenerator.cs

```diff
@@ -168,43 +168,43 @@ private Pattern generate()
                 }
 
                 if (convertType.HasFlag(PatternType.KeepSingle))
-                    return pattern = generateRandomNotes(1);
+                    return generateRandomNotes(1);
 
                 if (convertType.HasFlag(PatternType.Mirror))
                 {
                     if (ConversionDifficulty > 6.5)
-                        return pattern = generateRandomPatternWithMirrored(0.12, 0.38, 0.12);
+                        return generateRandomPatternWithMirrored(0.12, 0.38, 0.12);
                     if (ConversionDifficulty > 4)
-                        return pattern = generateRandomPatternWithMirrored(0.12, 0.17, 0);
+                        return generateRandomPatternWithMirrored(0.12, 0.17, 0);
 
-                    return pattern = generateRandomPatternWithMirrored(0.12, 0, 0);
+                    return generateRandomPatternWithMirrored(0.12, 0, 0);
                 }
 
                 if (ConversionDifficulty > 6.5)
                 {
                     if (convertType.HasFlag(PatternType.LowProbability))
-                        return pattern = generateRandomPattern(0.78, 0.42, 0, 0);
+                        return generateRandomPattern(0.78, 0.42, 0, 0);
 
-                    return pattern = generateRandomPattern(1, 0.62, 0, 0);
+                    return generateRandomPattern(1, 0.62, 0, 0);
                 }
 
                 if (ConversionDifficulty > 4)
                 {
                     if (convertType.HasFlag(PatternType.LowProbability))
-                        return pattern = generateRandomPattern(0.35, 0.08, 0, 0);
+                        return generateRandomPattern(0.35, 0.08, 0, 0);
```

Comment:
All of this changes the code that will be run in `finally`

Reply:
OK this should be another false positive about `finally`.

Reply:
I'm doing a minor refactor on this. Using `try` as control flow is strongly discouraged and somehow misleading. Since the method has only one caller, I've made it a local function and move post-process in `finally` to the caller.